pkg/operator/controller/machineconfig/mcp.go

```diff
@@ -168,15 +168,6 @@ func (r *MachineConfigReconciler) CheckNodeObservabilityMCPStatus(ctx context.Co
 
 	r.Lock()
 
-	if !r.CtrlConfig.Status.IsDebuggingEnabled() {
```

Comment:
Condition handled [here](https://github.com/openshift/node-observability-operator/commit/9d1ecfca265c761477ff0827b79b145ceff3eb5d), before calling the method, hence removed redundant check.